src/App/Pages/Accounts/HomePageViewModel.cs

```diff
@@ -125,18 +125,20 @@ public async Task ContinueToLoginStepAsync()
                         AppResources.Ok);
                     return;
                 }
+
                 await _stateService.SetRememberedEmailAsync(RememberEmail ? Email : null);
                 var userId = await _stateService.GetUserIdAsync(Email);
-                if (!string.IsNullOrWhiteSpace(userId))
+
+                if (!string.IsNullOrWhiteSpace(userId) &&
+                    (await _stateService.GetEnvironmentUrlsAsync(userId))?.Base == _environmentService.BaseUrl &&
+                    await _stateService.IsAuthenticatedAsync(userId))
+                {
+                    await _accountManager.PromptToSwitchToExistingAccountAsync(userId);
```

Comment:
Looks good, but per existing patterns leave the `return` here and keep `StartLoginAction()` outside like before